OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -244,6 +244,10 @@ public Mobile(ActorInitializer init, MobileInfo info)
 				SetVisualPosition(self, init.Get<CenterPositionInit, WPos>());
 
 			moveIntoWorldDelay = init.Contains<MoveIntoWorldDelayInit>() ? init.Get<MoveIntoWorldDelayInit, int>() : 0;
+
+			// Use -1 to indicate the activity should not run at all
```

Comment:
If we are going to make the meaning of `moveIntoWorldDelay = -1` be to not `MoveIntoWorld` at all, then we might as well just use `MoveIntoWorldDelayInit` for that instead of a new Init. Maybe the name can be altered slightly to make this more obvious. Otherwise I think we should just use a separate boolean here as well.

Reply:
I think its important to make the intent explicit in the external API, so I made `moveIntoWorldDelay` nullable instead.

Reply:
But why involve `moveIntoWorldDelay` at all then? Just use a separate `skipMoveIntoWorld` flag in that case.

Reply:
The value of moveIntoWorldDelay is meaningless if we are skipping the activity. I don’t see the benefit from adding a flag that says “the value in another variable is bogus” when we could encode that directly in the variable itself.